core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -116,6 +122,11 @@ public void initialize(String name, Map<String, String> properties) {
     }
   }
 
+  public void setFileIOBuilder(Function<Map<String, String>, FileIO> ioBuilder) {
```

Comment:
I looked into adding this to `BaseMetastoreCatalog` but it didn't make sense. That class is not `Configurable` and creating an IO requires a `Configuration` when the subclasses implement it. As a result, this class would need to override both this method and a `newFileIO` method so it isn't worth it.

Reply:
Seeing as how we're duplicating this API across some catalogs, should we pull the set/create into an interface?

Reply:
I'd like to see whether we need this more broadly first. It's a pretty narrow use case at the moment.